subsys/bluetooth/services/dis.c

```diff
@@ -45,6 +45,19 @@ static struct dis_pnp dis_pnp_id = {
 };
 #endif
 
+#if defined(CONFIG_BT_DIS_SYSTEM_ID)
+static uint8_t dis_system_id[8] = {
+	(CONFIG_BT_DIS_SYSTEM_ID_OUI >> 0) & 0xFF,
+	(CONFIG_BT_DIS_SYSTEM_ID_OUI >> 8) & 0xFF,
+	(CONFIG_BT_DIS_SYSTEM_ID_OUI >> 16) & 0xFF,
+	((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 0) & 0xFF,
+	((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 8) & 0xFF,
+	((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 16) & 0xFF,
+	((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 24) & 0xFF,
+	((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 32) & 0xFF
```

Comment:
We should use the macros from `bluetooth/byteorder.h`.

If we find it's supposed to be big-endian after all, you may optionally add and use new macros if you want.

Reply:
`byteorder.h` does not contain macros to store two number into the same array.

Reply:
`zephyr/bluetooth/byteorder.h` (not to be confused with `zephyr/sys/byteorder.h`) has macros that produce a comma-separated list of octets. For two numbers, we would use them separately for each number, like `BT_BYTES_LIST_BE24(system_id_oui), BT_BYTES_LIST_BE40(system_id_id)`.

I created a PR with big-endian macros: https://github.com/zephyrproject-rtos/zephyr/pull/77146

Reply:
I have now implemented the big-endian macros.